### ISSUE TITLE
UnicodeDecodeError during comment generation

### ISSUE THREAD

**Issue by jirikralik** [py-cov-action/python-coverage-comment-action#348]
Hello guys,

One of our PR is having this problem. I dont see in the code anything that would impact the generation (and it was even passing before a few commits ago). We just added some new .py files and documents. Here is the .coverage file. I would appreciate any help. Thanks!
[coverage.zip](https://github.com/py-cov-action/python-coverage-comment-action/files/13963884/coverage.zip)

This is the error:
```
Notice: Generating comment for PR
Error: Critical error. This error possibly occurred because the permissions of the workflow are set incorrectly. You can see the correct setting of permissions here: https://github.com/py-cov-action/python-coverage-comment-action#basic-usage
Otherwise please look for open issues or open one in https://github.com/py-cov-action/python-coverage-comment-action/
Traceback (most recent call last):
  File "/workdir/coverage_comment/main.py", line 46, in main
    exit_code = action(
                ^^^^^^^
  File "/workdir/coverage_comment/main.py", line 98, in action
    return process_pr(
           ^^^^^^^^^^^
  File "/workdir/coverage_comment/main.py", line 135, in process_pr
    added_lines = coverage_module.get_added_lines(git=git, base_ref=base_ref)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workdir/coverage_comment/coverage.py", line 286, in get_added_lines
    diff = git.diff("--unified=0", "FETCH_HEAD", "--", ".")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workdir/coverage_comment/subprocess.py", line 63, in _git
    return run(
           ^^^^
  File "/workdir/coverage_comment/subprocess.py", line 22, in run
    return subprocess.run(
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 550, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 1209, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 2146, in _communicate
    stdout = self._translate_newlines(stdout,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 1086, in _translate_newlines
    data = data.decode(encoding, errors)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe2 in position 2537696: invalid continuation byte
##[debug]Docker Action run completed with exit code 1
##[debug]Finishing: Coverage comment
```

This is the action:
```
backend-test:
    if: needs.files-changed.outputs.backend_src == 'true'
    needs: files-changed
    strategy:
      matrix:
        python: ["3.10"]
        platform: [ubuntu-latest]
    runs-on: ubuntu-20.04
    permissions:
      # Gives the action the necessary permissions for publishing new
      # comments in pull requests.
      pull-requests: write
      # Gives the action the necessary permissions for pushing data to the
      # python-coverage-comment-action branch, and for editing existing
      # comments (to avoid publishing multiple comments in the same PR)
      contents: write
    steps:
      - name: Self Hosted Runner Post Job Cleanup Action
        uses: TooMuch4U/actions-clean@v2.1
      - name: Checkout
        uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python }}
        uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python }}
      - name: Install test dependencies
        run: python -m pip install -U tox tox-docker
      - name: Run tests
        run: tox -p -m tests --parallel-no-spinner
      - name: 'Upload Artifact'
        uses: actions/upload-artifact@v3
        with:
          name: coverage
          path: .coverage
          retention-days: 1
      - name: Coverage comment
        id: coverage_comment
        uses: py-cov-action/python-coverage-comment-action@v3
        with:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by jirikralik** [comment]
it seems like the problem is connected with PDF files in our repo

**Comment by ewjoachim** [comment]
Hm, I would have guessed that Git was always clever enough to avoid dupming bytes that are not text to its output. I guess the issue is that not always.

I guess that if you run 
```
git diff --unified=0 <your_main_branch>..<your_pr_branch> -- .
```
it show you some binary symbols in the diff of your PDF ?
I guess we should handle git's output as `bytes` and not translate it to `str`.
Currently we're calling `subprocess.run(..., text=True, ...)`. I believe adding `errors='ignore'` (or `'replace'`) would ignore characters that can't be decoded (the name may give the impression we're ignoring all errors of the subprocess but it actually only relates to decoding the binary output)

Let's have a try :)

**Comment by jirikralik** [comment]
It is passing! Cool!

**Comment by ewjoachim** [comment]
Closed by #349
